packages/terra-date-input/src/DateInput.jsx

```diff
@@ -442,6 +468,32 @@ class DateInput extends React.Component {
     }
   }
 
+  /**
+   * Sets state to new value dependent on existing date
+   * @param {Object} event Event object generated from the event delegation.
+   * @param {Number} addDays Adds days to current date or today's date.
+   */
+  setHotKeyDate(event, addDays) {
+    const validYear = (this.state.year.length === 4 && Number(this.state.year) < DateInputUtil.MaxYearValue && Number(this.state.year) > DateInputUtil.MinYearValue);
+    const validDay = (Number(this.state.day) <= 31 && Number(this.state.day) > 0);
+    let dateObj;
+    if (!this.state.month || addDays === 0 || !validYear || !validDay) {
```

Comment:
If any of the year, day or month is invalid but not all three are empty (basically we don't have a complete date), and I press either `-` or `+`, it will get in here and create today's day. According to the requirements, we should ignore the -/+ key press and do nothing instead of continuing to add/subtract a day from today's date.

Reply:
After further discussion and clarification, pressing the `-` or `+` key when the date is incomplete should continue to increment/decrement from today's date. The above comment can be ignored.

For example, if month is `February` and day is `01` but the year is `20` which is incomplete, pressing `+` should set to tomorrow's date (today + 1).